### ISSUE TITLE
Have support for DataDog sink

### ISSUE THREAD

**Issue by anuprout** [tremor-rs/tremor-runtime#1214]
**Describe the problem you are trying to solve**
Support for writing data to DataDog using REST apis. 

**Describe the solution you'd like**

**Notes**

If this is an implementation of an [RFC](https://github.com/tremor-rs/tremor-rfcs) provide a URL
to the RFC this enhancement implements.

If this is a major enhancement or contribution an RFC may be required. It is ok to submit an enhancement
first and our core team will assist with major contributions. In general, major contributions should be
discussed with the community before submission.

**Comment by mfelsche** [comment]
Important info here:

> All requests to Datadog’s API must be authenticated. Requests that write data require reporting access and require an API key. Requests that read data require full access and also require an application key.

https://docs.datadoghq.com/api/latest/authentication/

A datadog connector needs to have an `api key` and an `application key` configured. As well as a `datadog_host` config.
We could pull this off by adding an auth config on the rest offramp/connector specialized for datadog.

The required format for metrics is:

```json
{
  "series": [
    {
      "host": "test.example.com",
      "interval": 20,
      "metric": "system.load.1",
      "points": [
        [
          1575317847,
          0.5
        ]
      ],
      "tags": [
        "environment:test"
      ],
      "type": "rate"
    }
  ]
}
```

So each event needs to be wrapped in a json object, which we could either support via the batch operator or a group by select statement.

But i think it is more convenient for users to have a ready-made `datadog-metrics` connector to just send stuff to.

**Comment by Licenser** [comment]
we can avoid a datadoc conector by setting this in a script

**Comment by mfelsche** [comment]
i know, this would solve the immediate pain of not being able to send stuff to datadog. So, in this sense, it is already possible right now. But having users go through all those hoops to send stuff to datadog is really inconvenient.

For the sake of usability, although it would duplicate code we have around here and there, i would suggest to create a `datadog` connector, instead of providing it the ikea way (bits and pieces to put together on your own).

**Comment by Licenser** [comment]
I would disagree, I think packaging this up with modular queries instead of extra connectors is in the long term a better usability. having hundreds of different connectors, is not as useful as having a few well polished  ones and pluggable pieces in between
